addons/mail/security/mail_security.xml

```diff
@@ -7,9 +7,10 @@
             <field name="model_id" ref="model_mail_channel"/>
             <field name="groups" eval="[Command.link(ref('base.group_user')), Command.link(ref('base.group_portal')), Command.link(ref('base.group_public'))]"/>
             <field name="domain_force">['|', '|',
-('public', '=', 'public'),
-'&amp;', ('public', '=', 'private'), ('is_member', '=', True),
-'&amp;', ('public', '=', 'groups'), ('group_public_id', 'in', [g.id for g in user.groups_id])]</field>
+                '&amp;', ('channel_type', '!=', 'channel'), ('is_member', '=', True),
+                '&amp;', ('channel_type', '=', 'channel'), ('group_public_id', '=', False),
+                '&amp;', ('channel_type', '=', 'channel'), ('group_public_id', 'in', [g.id for g in user.groups_id])]
```

Comment:
cleanup: instead of keeping the old format, just make one case for != channel, and one case for channel with the OR on group_public_id only inside it.